include/dali/core/mm/pool_resource.h

```diff
@@ -277,7 +178,7 @@ class pool_resource_base : public memory_resource<kind, Context> {
     if (options_.sync == sync_scope::device) {
       int prev = -1;
       const int kMaxDevices = 256;
-      uint32_t dev_mask[kMaxDevices >> 5];  // NOLINT - linter doesn't notice it's a constant expression
+      uint32_t dev_mask[kMaxDevices >> 5] = {};  // NOLINT - linter doesn't notice it's a constant expression
```

Comment:
{} - the array was not initialized, synchronization might have been missed!

Reply:
would std::array have helped here?

Reply:
No, it's not initialized either.